### PR TITLE
Add activity feeds API

### LINKED ISSUE #5666: [API] Activity API
Hi,

Is there an API for activity/dashboard view?

I also noticed that there is `private` variable used in `api/v1/user/repos` API for private/public access type. In java it is reserved keyword and cannot be used. Is there a way around it?. I need it for showing lock icon.

Thanks

### PR DESCRIPTION
**Pull request by Zettat123**
Close #5666

Add APIs for getting activity feeds.

### CODE REVIEW

File: modules/structs/activity.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package structs
+
+import "time"
+
+type Activity struct {
+	ID        int64       `json:"id"`
+	UserID    int64       `json:"user_id"` // Reiciver user
```
Comment (Konctantin) [inline modules/structs/activity.go]: Receiver user

**Review by silverwind — APPROVED** [review]

File: routers/api/v1/org/org.go [hunk 2]
```diff
@@ -370,3 +371,69 @@ func Delete(ctx *context.APIContext) {
...
+		if ctx.Doer.IsAdmin {
+			includePrivate = true
+		} else {
+			org := organization.OrgFromUser(ctx.ContextUser)
+			isMember, err := org.IsOrgMember(ctx.Doer.ID)
+			if err != nil {
+				ctx.Error(http.StatusInternalServerError, "IsOrgMember", err)
+				return
+			}
+			includePrivate = isMember
```
Comment (lunny) [inline routers/api/v1/org/org.go:417]: This is not right check. Different repositories may have different permissions check. You need to check team repositories and permissions.

Comment (Zettat123) [reply routers/api/v1/org/org.go:417]: Repository accessibility will be checked by the following code 

https://github.com/go-gitea/gitea/blob/d67e40684f43b0eb744cad26e0265002f033dbc3/models/activities/action.go#L445-L447

**Review by lunny — APPROVED** [review]

File: models/activities/action.go [hunk 3]
```diff
@@ -66,6 +66,67 @@ const (
...
+	case ActionPublishRelease:
+		return "PUBLISH_RELEASE"
+	case ActionPullReviewDismissed:
+		return "PULL_REVIEW_DISMISSED"
+	case ActionPullRequestReadyForReview:
+		return "PULL_REQUEST_READY_FOR_REVIEW"
+	case ActionAutoMergePullRequest:
+		return "AUTO_MERGE_PULL_REQUEST"
+	default:
+		return strconv.Itoa(int(at))
```
Comment (wxiaoguang) [inline models/activities/action.go]:
```suggestion
		return "ACTION-" + strconv.Itoa(int(at))
```

ps: why there are all UPPERCASE ?  I guess this is the first time in the codebase to use whole uppercase symbols.

File: routers/api/v1/api.go [hunk 4]
```diff
@@ -1172,6 +1176,10 @@ func Routes(ctx gocontext.Context) *web.Route {
 				m.Get("/issue_config", context.ReferencesGitRepo(), repo.GetIssueConfig)
 				m.Get("/issue_config/validate", context.ReferencesGitRepo(), repo.ValidateIssueConfig)
 				m.Get("/languages", reqRepoReader(unit.TypeCode), repo.GetLanguages)
+
+				m.Group("/activities", func() {
+					m.Get("/feeds", repo.ListRepoActivityFeeds)
+				})
```
Comment (wxiaoguang) [inline routers/api/v1/api.go]:
```suggestion
				m.Group("/activities/feeds", repo.ListRepoActivityFeeds)
```

Here and below, like this?

File: models/activities/action.go (hunk 3 above)
Comment (Zettat123) [reply models/activities/action.go]: You are right. It looks strange so I changed them to lower case.

File: models/activities/action.go [hunk 5]
```diff
@@ -66,6 +66,67 @@ const (
...
+	case ActionPublishRelease:
+		return "publish_release"
+	case ActionPullReviewDismissed:
+		return "pull_review_dismissed"
+	case ActionPullRequestReadyForReview:
+		return "pull_request_ready_for_review"
+	case ActionAutoMergePullRequest:
+		return "auto_merge_pull_request"
+	default:
+		return "action" + strconv.Itoa(int(at))
```
Comment (wxiaoguang) [inline models/activities/action.go]:
```suggestion
		return "action-" + strconv.Itoa(int(at))
```

add a separator.

**Pull request merged**